app/actors/curation_concerns/actors/apply_order_actor.rb

```diff
@@ -20,10 +19,14 @@ def sync_members(ordered_member_ids)
 
           (ordered_member_ids - existing_members_ids).each do |work_id|
             work = ::ActiveFedora::Base.find(work_id)
-            curation_concern.ordered_members << work
+            if work.depositor != curation_concern.depositor
+              curation_concern.errors[:ordered_member_ids] << "Works can only be related to each other if they were deposited by the same user."
```

Comment:
Doesn't matter where it goes, we still need to encapsulate what users can do in the Ability class. That gives downstream users a single place to change what a user can/cannot do.